### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by AnthonyLim23**
I have added the docs for quickBayes. These give users the information needed to understand the code base and to work with it as both a package and as a developer. 

To test read the docs using the read the docs build below.

### CODE REVIEW

File: docs/source/fitting/fit_engines.rst [hunk 1]
```diff
@@ -1,4 +1,69 @@
...
+    engine.do_fit(x, y, e, g_func2)
+    chi_2 = engine.get_chi_squared()
+    params = engine.get_fit_parameters()
+    print(chi_2, params)
+
+
+GoFit
+=====
+
+The :code:`GoFitEngine` uses the :code:`multistart` method from GoFit.
```
Comment (jess-farmer) [inline docs/source/fitting/fit_engines.rst:60]: May be worth adding a link here to the GoFit repo/docs?

File: docs/source/fitting/fit_functions.rst [hunk 2]
```diff
@@ -1,4 +1,91 @@
 Fit Functions
 =============
 
-fit functions are fun
+Fit functions are the mathematical representations of the data.
+In quickBayes the fit functions are all classes with a very specific structure, see <insert link> for a full list of currently available options.
```
Comment (jess-farmer) [inline docs/source/fitting/fit_functions.rst:5]: Missing link needs adding in here

File: docs/source/fitting/fit_functions.rst [hunk 3]
```diff
@@ -1,4 +1,91 @@
...
+.. code-block:: python
+
+  params = function.read_from_report(report)
+  print(params)
+
+Hence,
+
+.. code-block:: python
+
+  [1.0, 2.0, 3.0]
```
Comment (jess-farmer) [inline docs/source/fitting/fit_functions.rst:63]: Should this be `[1.0, 0, 2.0]`?

File: docs/source/loglikelihood.rst [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+- The fit for the model is at a minima.
+- The model can represent the data.
+
+"Data Analysis A bayesian tutorial second edition", by D. S. Sivia (equation 4.20 page 88) uses the above assumptions to show that the probability can be written as
 
 .. math::
     P(M|D_k,I) \propto \frac{N! (4\pi)^N}{\beta^N\sqrt{\det(H)}}\exp{(-\frac{\chi^2}{2})},
 
-where :math:`\\beta = (x_{max}-x_{min})A_{max}`, :math:`H` is the Hessian matrix. We want this as a loglikelihood, so take :math:`\\log_{10}` and use:
+where :math:`\\beta = (x_{max}-x_{min})A_{max}`, :math:`H` is the Hessian matrix.
```
Comment (jess-farmer) [inline docs/source/loglikelihood.rst:15]:
```suggestion
where :math:`\beta = (x_{max}-x_{min})A_{max}`, :math:`H` is the Hessian matrix.
```
There's a couple of points on this page where there's an extra backslash and math symbols are being shown as words

File: docs/source/workflows/workflows.rst [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+- :code:`execute` for doing the analysis.
+
+
+Model Selection
+===============
+
+The main advantage of a workflow is to simplify model selection.
+Examples for the models being selected between could be the number of Lorentzians (:code:`QLData`) or the number of exponential decays (:code:`MuonExpDecay`) .
+The :code:`ModelSelectionWorkflow` simplifies creating these kind of analyses.
+It inherits the the :code:`workflowTemplate` class but has as additional property :code`get_parameters_and_errors`.
```
Comment (jess-farmer) [inline docs/source/workflows/workflows.rst:72]:
```suggestion
It inherits the the :code:`workflowTemplate` class but has as additional property :code:`get_parameters_and_errors`.
```

File: docs/source/history.rst [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+The quickBayes packages started as a replacement for the quasielasticbayes package, which was a Fortran code base that was exposed to Python.
+In the development of quickBayes the code became more modular and flexable. 
+This allows the code to be applied to any probelm that requires model selection. 
+
+To verify the results of quickBayes it was compared to the quasielasticbayes calculations. 
+For lorentzian fits it produced the close enough results.
+However, for a stretched exponential it gave different results for the FWHM.
+The stretched exponential results for the FWHM are different to quasielasticbayes, as shown by the figure below.
+However, the new results agree with the FWHM values for fitting a single Lorentzian ('QL').
+The quickBayes method provides FWHM results that are comparable for all :math:`Q` values (green and black data), unlike the original code that has a divergence for low :math`Q` value.
```
Comment (jess-farmer) [inline docs/source/history.rst:13]:
```suggestion
The quickBayes method provides FWHM results that are comparable for all :math:`Q` values (green and black data), unlike the original code that has a divergence for low :math:`Q` value.
```

**Review by jess-farmer — CHANGES_REQUESTED** [review]
Looks great - just a couple of very minor comments

File: docs/source/fitting/fit_engines.rst (hunk 1 above)
Comment (AnthonyLim23) [inline docs/source/fitting/fit_engines.rst:60]:
```suggestion
The :code:`GoFitEngine` uses the :code:`multistart` method from `GoFit <https://ralna.github.io/GOFit/_build/html/index.html>`.
```

File: docs/source/fitting/fit_functions.rst (hunk 2 above)
Comment (AnthonyLim23) [reply docs/source/fitting/fit_functions.rst:5]:
```suggestion
In quickBayes the fit functions are all classes with a very specific structure, see :ref:`here<quickBayes.fit_functions
>` for a full list of currently available options.
```

File: docs/source/fitting/fit_functions.rst (hunk 3 above)
Comment (AnthonyLim23) [inline docs/source/fitting/fit_functions.rst:63]:
```suggestion
  [1.0, 0.0, 2.0]
```

File: docs/source/loglikelihood.rst [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
+The :math:`\\beta` value represents the variation in the data by taking the product of the length of the :math:`x` range with the maximum height of the data.
```
Comment (AnthonyLim23) [inline docs/source/loglikelihood.rst:16]:
```suggestion
The :math:`\beta` value represents the variation in the data by taking the product of the length of the :math:`x` range with the maximum height of the data.
```

File: docs/source/loglikelihood.rst [hunk 8]
```diff
@@ -1,14 +1,20 @@
...
+To convert the probability into a loglikelihood we need to take the :math:`\\log_{10}` and use:
```
Comment (AnthonyLim23) [inline docs/source/loglikelihood.rst:17]:
```suggestion
To convert the probability into a loglikelihood we need to take the :math:`\log_{10}` and use:
```

**Review by jess-farmer — APPROVED** [review]

**Pull request merged**